### ISSUE TITLE
gdtools SUBTRACT fatal segmentation fault error 

### ISSUE THREAD

**Issue by vr1087** [barricklab/breseq#251]
I am seeing a segmentation fault when subtracting mutations with gdtools. No problems were found in the files when checked with gdtools VALIDATE and normalizing the files with gdtools NORMALIZE did not resolve the issue. 

The issue might lie in something I did, or in the gdtools INTERSECT command as I only seem to get this error when subtracting intersected genomeDiff files.

I am using:

```
$ gdtools --version
gdtools 0.35.4
```

Steps to re-create error:

```
$ gdtools INTERSECT -o intersection.gd i_left.gd i_right.gd
$ gdtools SUBTRACT -o result.gd intersection.gd subtract.gd
================================================================================
breseq 0.35.4  revision f352f80f4bc9   http://barricklab.org/breseq

Active Developers: Barrick JE, Deatherage DE
Contact:           <jeffrey.e.barrick@gmail.com>

breseq is free software; you can redistribute it and/or modify it under the
terms the GNU General Public License as published by the Free Software 
Foundation; either version 2, or (at your option) any later version.

Copyright (c) 2008-2010 Michigan State University
Copyright (c) 2011-2017 The University of Texas at Austin

If you use breseq in your research, please cite:

  Deatherage, D.E., Barrick, J.E. (2014) Identification of mutations
  in laboratory-evolved microbes from next-generation sequencing
  data using breseq. Methods Mol. Biol. 1151: 165–188.

If you use structural variation (junction) predictions, please cite:

  Barrick, J.E., Colburn, G., Deatherage D.E., Traverse, C.C.,
  Strand, M.D., Borges, J.J., Knoester, D.B., Reba, A., Meyer, A.G. 
  (2014) Identifying structural variation in haploid microbial genomes 
  from short-read resequencing data using breseq. BMC Genomics 15:1039.
================================================================================

*** Begin SUBTRACT ***
    Starting with input GD file: intersection.gd

    Subtracting mutations from files
        subtract.gd
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!> FATAL ERROR <!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
Segmentation Fault
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!> STACK TRACE <!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
Backtrace with 8 stack frames.
gdtools() [0x45dbb6]
gdtools() [0x45dde5]
/lib/x86_64-linux-gnu/libc.so.6(+0x3efd0) [0x7f960501dfd0]
gdtools() [0x4bfa9e]
gdtools() [0x41c068]
gdtools() [0x40ae21]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7f9605000b97]
gdtools() [0x418d29]
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

PS. The gdfiles were created with breseq-0.35.1. I also got this segmentation fault while using gdtools 0.35.1.

**Comment by vr1087** [comment]
I am able to provide the gd files used in the example.

**Comment by jeffreybarrick** [comment]
Please send to my email in the _breseq_ header. Thanks!

**Comment by jeffreybarrick** [comment]
This is fixed in 06cea1014f1f0b3505e5d14254fba0734b68938f. Thanks for the test data!
